lib/ocamlorg/module_map.mli

```diff
@@ -0,0 +1,35 @@
+module StringMap : Map.S with type key = string
+
+type kind =
+  | Module
+  | Page
+  | LeafPage
+  | ModuleType
+  | Argument
+  | Class
+  | ClassType
+  | File
+
+module Module : sig
```

Comment:
Will we use the `Module` module here? If so, it would make more sense to me to have the module `Module` contain a module `Map`. What do you think?

Reply:
What do you mean ? Having something like this ?

``` ocaml
module Module : sig
  type t

  val name : t -> string

  val parent : t -> t option

  val submodules : t -> t StringMap.t

  val kind : t -> kind

  val path : t -> string

  module Map : sig
    type library =
      { name : string
      ; dependencies : string list
      ; modules : t StringMap.t
      }

    type t = { libraries : library StringMap.t }

    val of_yojson : Yojson.Safe.t -> t
  end
end

```

Reply:
Yes, that's what I meant